src/useCases/CreateEventStreamUseCase.ts

```diff
@@ -0,0 +1,35 @@
+import crypto from 'crypto';
+import { pipe } from 'fp-ts/lib/function';
+import * as E from 'fp-ts/Either';
+import * as TE from 'fp-ts/TaskEither';
+import { authorizeApiKey } from '../domain/authorize';
+import {
+  CreateEventStreamRecord,
+  CreateEventStreamRecordRepository,
+} from '../domain/CreateEventStreamRecordRepository';
+import { ApiKey } from '../generated/definitions/ApiKey';
+import { StreamCreationRequest } from '../generated/streams/StreamCreationRequest';
+
+export const CreateEventStreamUseCase =
+  (
+    repository: CreateEventStreamRecordRepository,
+    streamIdGenerator: () => string = () => crypto.randomUUID(),
+    nowDate: () => Date = () => new Date()
```

Comment:
```suggestion
    streamIdGenerator: IO<string> = () => crypto.randomUUID(),
    nowDate: IO<Date> = () => new Date()
```
just an alternative, not really a request change. but, if you are going to accept this, you should also import `IO`